config/rbac/role.yaml

```diff
@@ -16,6 +16,12 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
+  - appstudio.redhat.com
+  resources:
+  - applications/finalizers
+  verbs:
+  - update
```

Comment:
In my opinion, standalone `update` does not suffice for most of the cases and it would be better to align with the `application` by adding `get`, `create`, `patch`, `list`... (actually everything other than `delete`)

Reply:
I would have needed other permissions if my intent was to have the controller actually manage finalizers. It isn't though. Its just that this particular permission is needed to enable creating other resources that reference the application as their owner and have `controller` or `blockOwnerDeletion` set to true.

Following the principle of least privileges, I would rather not add additional privileges unless I can find a reason to have to.

Reply:
Sounds good to me